src/main/java/com/bringframework/factory/AutowiredBeanPostProcessor.java

```diff
@@ -34,14 +30,15 @@ public void process() {
   }
 
   private void processBean(String beanName, Object bean) {
-    var beanDefinition = definitionRegistry.getBeanDefinition(beanName);
-    log.debug("Injecting autowired field dependencies for bean {}", beanName);
-    beanDefinition.getAutowiredFieldsMetadata().values()
-        .forEach(field -> populateField(field, bean));
+    if (definitionRegistry.getAllBeanDefinitions().containsKey(beanName)) {
```

Comment:
I do have concerns here, because every time by calling `definitionRegistry.getAllBeanDefinitions()` new map is created and it is not efficient.
would just `definitionRegistry.getBeanDefinition(beanName)` work for us ?
then either check on null, or let definitionRegistry.getBeanDefinition method throw NoSuchBeanDefinitionException

Reply:
added method `contains(beanName)` to `definitionRegistry`. It should resolve the problem and shouldn't brake any existing and probably already tested logic